qubes/storage/file.py

```diff
@@ -265,9 +265,7 @@ def commit(self):
         return self
 
     def export(self):
-        # FIXME: this should rather return snapshot(self.path, self.path_cow)
-        #  if domain is running
-        return self.path
+        return snapshot(self.path, self.path_cow)
```

Comment:
The `snapshot` is not a real function. It's a pseudo-code for assembling dm-snapshot target (or rather - dm-snapshot-origin). There are two difficulties with this:
1. The actual device mapper is assembled in in `block-origin` script, which is called via libvirt -> libxl. It (in its current shape) can't be called outside because for example it parameters in xenstore.
2. Such device mapper needs to be cleaned up after use. Until recently, we didn't have API to cleanup device returned from `export` function (now we have it: `export_end`).

Doing this properly requires quite a bit of refactoring - at least around `block-snapshot`/`block-origin` scripts. They should really be a python code in this very file, instead of external shell scripts and libvirt should get already assembled device. But that's quite a bit of work that may not be worth for a storage driver we are going to remove (at least that was initial plan).
Do you see any use case that would warrant actively maintaining this driver? If so, it may be worth doing this refactor. If not, I'd rather prefer a band-aid blocking export on a running VM. If I'm not mistaken, export is used in backup code only.

Reply:
I honestly cannot think of any reason to keep maintaining it, and I suggest formally deprecating it.  The main reasons other virtualization tools support files are portability and not requiring root privileges, neither of which apply here.